ui/hooks/src/use-follow.ts

```diff
@@ -49,27 +59,46 @@ export function useIsFollowingMultiple(
   followingEthAddressArray: string[],
 ) {
   const queryClient = useQueryClient();
-  return useQuery(
+  const [refetchPending, setRefetchPending] = React.useState(false);
+  const filteredFollowingEthArray = followingEthAddressArray.filter(_ => _); // often we end up with [undefined] as arg
+  const allAddresses: string[] = queryClient.getQueryData([FOLLOWED_PROFILES_KEY, 'all']) || [];
+  const newAllAddressess = deduplicateArray([...allAddresses, ...filteredFollowingEthArray]);
+  const shouldRefetch = objHash(newAllAddressess) !== objHash(allAddresses);
```

Comment:
keep in mind that `objHash` is kinda slow for large data sets

Reply:
I thought of just doing .join('') but those could potentially get pretty big 🤔 Possibly just join the last n characters of each address, if this becomes a problem?